pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java

```diff
@@ -178,6 +178,11 @@ public KafkaRecord(ConsumerRecord<String,?> record, V value, Schema<V> schema) {
             return Optional.of(Integer.toString(record.partition()));
         }
 
+        @Override
+        public Optional<Integer> getPartitionNumber() {
+            return Optional.of(record.partition());
```

Comment:
Optional.ofNullable ?

Reply:
`record.partition()` returns int, cannot be null